ash/src/vk/macros.rs

```diff
@@ -20,17 +20,29 @@ macro_rules! vk_bitflags_wrapped {
                 self.0
             }
             #[inline]
-            pub fn is_empty(self) -> bool {
-                self == Self::empty()
+            pub const fn is_empty(self) -> bool {
+                self.0 == Self::empty().0
             }
             #[inline]
-            pub fn intersects(self, other: Self) -> bool {
-                self & other != Self::empty()
+            pub const fn intersects(self, other: Self) -> bool {
+                !Self(self.0 & other.0).is_empty()
             }
             #[doc = r" Returns whether `other` is a subset of `self`"]
             #[inline]
-            pub fn contains(self, other: Self) -> bool {
-                self & other == other
+            pub const fn contains(self, other: Self) -> bool {
+                self.0 & other.0 == other.0
+            }
+            #[inline]
+            pub const fn intersection(self, other: Self) -> Self {
```

Comment:
The `bitflags` crate calls this [`intersects`](https://docs.rs/bitflags/latest/bitflags/example_generated/struct.Flags.html#method.intersects). I think we should use the same name.